File: 0929-unique-email-addresses/0929-unique-email-addresses.js
/**
 * @param {string[]} emails
 * @return {number}
 */
var numUniqueEmails = function(emails) {
    let uniqueEmails = new Set();
    emails.forEach(email => {
        let parts = email.split('@');
        let local = parts[0];
        let domain = parts[1];
        local = local.split('+')[0];
        local = local.replace(/\./g, '');
        email = local + "@" + domain;
        uniqueEmails.add(email);
    });
    return uniqueEmails.size;
};

File: 0929-unique-email-addresses/0929-unique-email-addresses.py
class Solution:
    def numUniqueEmails(self, emails: List[str]) -> int:
        unique_emails = set()
        for email in emails:
            local_name, domain_name = email.split('@')
            local_name = local_name.replace('.', '')   # Remove all dots
            local_name = local_name.split('+')[0]      #  Select first item and ignore all text after +
            email = f"{local_name}@{domain_name}"
            unique_emails.add(email)

        return len(unique_emails)
